tests/loader_phys_dev_inst_ext_tests.cpp

```diff
@@ -5246,4 +5246,155 @@ TEST(LoaderInstPhysDevExts, GetDrmDisplayEXTMixed) {
+        inst.functions->vkGetInstanceProcAddr(inst, "vkGetPhysicalDeviceExternalSemaphorePropertiesKHR"));
+    auto GetPhysicalDeviceExternalFenceProperties = reinterpret_cast<PFN_vkGetPhysicalDeviceExternalFencePropertiesKHR>(
+        inst.functions->vkGetInstanceProcAddr(inst, "vkGetPhysicalDeviceExternalFencePropertiesKHR"));
+    ASSERT_NE(nullptr, GetPhysicalDeviceExternalBufferProperties);
+    ASSERT_NE(nullptr, GetPhysicalDeviceExternalSemaphoreProperties);
+    ASSERT_NE(nullptr, GetPhysicalDeviceExternalFenceProperties);
+
+    // The above are instance extensions, so shouldn't crash even if only one physical device supports each
+    // extension.
+    for (uint32_t dev = 0; dev < device_count; ++dev) {
+        VkPhysicalDeviceExternalBufferInfo ext_buf_info{};
+        VkExternalBufferProperties ext_buf_props{};
+        VkPhysicalDeviceExternalSemaphoreInfo ext_sem_info{};
+        VkExternalSemaphoreProperties ext_sem_props{};
+        VkPhysicalDeviceExternalFenceInfo ext_fence_info{};
+        VkExternalFenceProperties ext_fence_props{};
+        GetPhysicalDeviceExternalBufferProperties(physical_devices[dev], &ext_buf_info, &ext_buf_props);
+        GetPhysicalDeviceExternalSemaphoreProperties(physical_devices[dev], &ext_sem_info, &ext_sem_props);
+        GetPhysicalDeviceExternalFenceProperties(physical_devices[dev], &ext_fence_info, &ext_fence_props);
+    }
+}
+
+TEST(LoaderInstPhysDevExts, DifferentPhysicalDeviceExtensions) {
+    FrameworkEnvironment env{};
+
+    // Add 3 drivers each of which supports a different physical device extension
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2, VK_API_VERSION_1_0));
+    env.get_test_icd(0).physical_devices.push_back({"pd0", 7});
+    env.get_test_icd(0).physical_devices.back().extensions.push_back({VK_KHR_PERFORMANCE_QUERY_EXTENSION_NAME, 0});
+
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2, VK_API_VERSION_1_0));
+    env.get_test_icd(1).physical_devices.push_back({"pd1", 0});
+    env.get_test_icd(1).physical_devices.back().extensions.push_back({VK_EXT_SAMPLE_LOCATIONS_EXTENSION_NAME, 0});
+
+    env.add_icd(TestICDDetails(TEST_ICD_PATH_VERSION_2, VK_API_VERSION_1_0));
+    env.get_test_icd(2).physical_devices.push_back({"pd2", 1});
+    env.get_test_icd(2).physical_devices.back().extensions.push_back({VK_EXT_CALIBRATED_TIMESTAMPS_EXTENSION_NAME, 0});
+
+    DebugUtilsLogger log{VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT};
+    InstWrapper inst{env.vulkan_functions};
+    FillDebugUtilsCreateDetails(inst.create_info, log);
+    inst.CheckCreate();
+
+    const uint32_t expected_device_count = 3;
+    uint32_t device_count = expected_device_count;
+    auto physical_devices = std::vector<VkPhysicalDevice>(expected_device_count);
+    ASSERT_EQ(VK_SUCCESS, inst->vkEnumeratePhysicalDevices(inst, &device_count, physical_devices.data()));
+    ASSERT_EQ(device_count, expected_device_count);
+    handle_assert_has_values(physical_devices);
```

Comment:
A little helpr to make writing tests easier.
```suggestion
auto physical_devices = inst.GetPhysDevs(3);
```